### PR TITLE
fixes #14265 - import content hosts with hypervisors and guests

### PR DESCRIPTION
**Pull request by thomasmckay**
Dependent on https://github.com/Katello/katello/pull/5831

### CODE REVIEW

File: lib/hammer_cli_csv/content_hosts.rb [hunk 1]
```diff
@@ -159,101 +159,57 @@ def import_locally
...
-            @hypervisor_guests[@existing[line[HOST]]] << @existing[name]
+            @hypervisor_guests[@existing[line[HOST]]] << "#{line[ORGANIZATION]}/#{name}"
           end
 
-          update_host_facts(host_id, line)
-          update_host_collections(host_id, line)
-          update_subscriptions(host_id, line)
+          #update_host_facts(host, line)
+          #update_host_collections(host, line)
+          #update_subscriptions(host, line)
```
Comment (beav) [inline lib/hammer_cli_csv/content_hosts.rb]: I say just remove the code if you want it to go away, we can look at the git log to add it back later if needed.

**Comment by thomasmckay** [comment]
@beav - all working again. tested against nightly and Sat-6.2

File: lib/hammer_cli_csv/content_hosts.rb [hunk 2]
```diff
@@ -25,111 +25,53 @@ def export
         CSV.open(option_file || '/dev/stdout', 'wb', {:force_quotes => false}) do |csv|
           csv << [NAME, ORGANIZATION, ENVIRONMENT, CONTENTVIEW, HOSTCOLLECTIONS, VIRTUAL, HOST,
                   OPERATINGSYSTEM, ARCHITECTURE, SOCKETS, RAM, CORES, SLA, PRODUCTS, SUBSCRIPTIONS]
-          if @server_status['release'] == 'Headpin'
-            export_sam csv
-          else
-            export_foretello csv
-          end
+          export_katello csv
```
Comment (beav) [inline lib/hammer_cli_csv/content_hosts.rb]: :+1:

File: lib/hammer_cli_csv/content_hosts.rb [hunk 3]
```diff
@@ -25,111 +25,53 @@ def export
...
-            operatingsystem = "#{host['facts']['distribution.name']} " if host['facts']['distribution.name']
-            operatingsystem += host['facts']['distribution.version'] if host['facts']['distribution.version']
-            architecture = host['facts']['uname.machine']
-            sockets = host['facts']['cpu.cpu_socket(s)']
-            ram = host['facts']['memory.memtotal']
-            cores = host['facts']['cpu.core(s)_per_socket'] || 1
+            virtual = host['facts']['virt::is_guest'] == 'true' ? 'Yes' : 'No'
+            hypervisor_host = host['subscription_facet_attributes']['virtual_host'].nil? ? nil : host['subscription_facet_attributes']['virtual_host']['name']
+            operatingsystem = "#{host['facts']['distribution::name']} " if host['facts']['distribution::name']
+            operatingsystem += host['facts']['distribution::version'] if host['facts']['distribution::version']
```
Comment (beav) [inline lib/hammer_cli_csv/content_hosts.rb]: you may want the space to precede `distribution::version`, that way it's only added when needed.

Also I would declare as `' ' + host['facts']['distribution::version']` just to make it really obvious that a space is being added

File: lib/hammer_cli_csv/content_hosts.rb [hunk 4]
```diff
@@ -305,44 +215,65 @@ def os_name_version(operatingsystem)
...
+      def update_existing(line)
+        if !@existing[line[ORGANIZATION]]
+          @existing[line[ORGANIZATION]] = true
+          # Fetching all content hosts can be too slow and times so page
+          # http://projects.theforeman.org/issues/6307
+          total = @api.resource(:hosts).call(:index, {
+              'organization_id' => foreman_organization(:name => line[ORGANIZATION]),
+              'per_page' => 1
+          })['total'].to_i
+          (total / 20 + 1).to_i.times do |page|
```
Comment (beav) [inline lib/hammer_cli_csv/content_hosts.rb]: maybe break out `per_page` to a var since you use it 2x.

**Comment by beav** [comment]
small updates but seems reasonable overall

File: lib/hammer_cli_csv/content_hosts.rb (hunk 4 above)
Comment (thomasmckay) [inline lib/hammer_cli_csv/content_hosts.rb]: leaving as explicit value since the two uses are different

**Comment by beav** [comment]
ACK

**Pull request merged**